manifests/port.pp

```diff
@@ -36,20 +36,17 @@
 #  }
 #
 define selinux::port (
-  $context  = undef,
+  $context,
+  $port,
   $protocol = undef,
-  $port     = undef,
-  $policy   = 'targeted'
 ) {
-  Exec {
-    path => '/bin:/sbin:/usr/bin:/usr/sbin',
-  }
 
-  if ( $context == undef ) or ( $port == undef ) {
```

Comment:
The error that puppet throws is that "context is required" or "port is required".  This is how I've seen all of the large modules enforce required parameters.